pkg/controller/clone-controller.go

```diff
@@ -304,10 +299,20 @@ func (r *CloneReconciler) updatePvcFromPod(sourcePod *corev1.Pod, pvc *corev1.Pe
 		setConditionFromPodWithPrefix(pvc.Annotations, AnnSourceRunningCondition, sourcePod)
 	}
 
+	var err error = nil
 	if !reflect.DeepEqual(currentPvcCopy, pvc) {
-		return r.updatePVC(pvc)
+		err = r.updatePVC(pvc)
 	}
-	return nil
+
+	if err == nil && podSucceededFromPVC(pvc) && pvc.Annotations[AnnCloneOf] != "true" {
+		// Now PVC has been fully updated and cloning can be completed.
+		log.V(1).Info("Adding CloneOf annotation to PVC")
+		pvc.Annotations[AnnCloneOf] = "true"
+		r.recorder.Event(pvc, corev1.EventTypeNormal, CloneSucceededPVC, cloneComplete)
+
+		err = r.updatePVC(pvc)
```

Comment:
Sure you need a second updatePVC in this case?

Reply:
Also I don't think this will solve the problem, we have not refreshed the PVC, so any attempt to update again will fail again for the same reason.
I think changing https://github.com/kubevirt/containerized-data-importer/blob/main/pkg/controller/clone-controller.go#L188-L193 to something like this should solve the problem:
```go
if r.hasFinalizer(pvc, cloneSourcePodFinalizer) && pvc.Annotations[AnnCloneOf] == "true" {
// Clone completed, remove source pod and finalizer.
  if err := r.cleanup(pvc, log); err != nil {
    return reconcile.Result{}, err
  }
}
```
This way we don't delete the source pod unless we know the annotation is set on the PVC. So if the update fails, it will just re-run the reconcile, and the pod hasn't been deleted.

Reply:
We probably want to extract the 'isPVCFullyUpdated' check into a function. So we can put all the logic associated with that into that function.

Reply:
That wouldn't solve it either. The problem is that we add `AnnCloneOf` when the source pod is still in Running phase, and then we delete it without updating PVC annotations.